tests/e2e/run-scanner-v4.bats

```diff
@@ -0,0 +1,175 @@
+    export_test_environment
+    if [[ "$CI" = "true" ]]; then
+        setup_gcp
+        setup_deployment_env false false
+    fi
+
+    if (( test_case_no == 0 )); then
+        # executing initial teardown to begin test execution in a well-defined state
+       run remove_existing_stackrox_resources
+    fi
+    test_case_no=$(( test_case_no + 1))
+}
+
+teardown() {
+    run remove_existing_stackrox_resources
+}
+
+@test "Upgrade from old Helm chart to HEAD Helm chart with Scanner v4 enabled" {
+    local MAIN_IMAGE_TAG=""
+
+    if [[ "$CI" = "true" ]]; then
+        setup_default_TLS_certs
+    fi
+
+    # Deploy earlier version without Scanner V4.
+    local _CENTRAL_CHART_DIR_OVERRIDE="${CHART_REPOSITORY}${CHART_BASE}/${EARLIER_CHART_VERSION}/central-services"
+    info "Deplying StackRox services using chart ${_CENTRAL_CHART_DIR_OVERRIDE}"
+    if [[ -n "${EARLIER_MAIN_IMAGE_TAG:-}" ]]; then
+        MAIN_IMAGE_TAG=$EARLIER_MAIN_IMAGE_TAG
+        info "Overriding MAIN_IMAGE_TAG=$EARLIER_MAIN_IMAGE_TAG"
+    fi
+    (
+        # shellcheck disable=SC2030,SC2031
+        export MAIN_IMAGE_TAG
+        # shellcheck disable=SC2030,SC2031
+        export ROX_SCANNER_V4_ENABLED=true
+        # shellcheck disable=SC2030,SC2031
+        export CENTRAL_CHART_DIR_OVERRIDE="${_CENTRAL_CHART_DIR_OVERRIDE}"
+        # shellcheck disable=SC2030,SC2031
+        export OUTPUT_FORMAT=helm
+        deploy_stackrox >&3
+    )
+
+    # Upgrade to HEAD chart without explicit disabling of Scanner v4.
+    info "Upgrading StackRox using HEAD Helm chart"
+    MAIN_IMAGE_TAG=""
+    if [[ -n "${CURRENT_MAIN_IMAGE_TAG:-}" ]]; then
+        MAIN_IMAGE_TAG=$CURRENT_MAIN_IMAGE_TAG
+        info "Overriding MAIN_IMAGE_TAG=$CURRENT_MAIN_IMAGE_TAG"
+    fi
+    (
+        # shellcheck disable=SC2030,SC2031
+        export MAIN_IMAGE_TAG
+        # shellcheck disable=SC2030,SC2031
+        export ROX_SCANNER_V4_ENABLED=true
+        # shellcheck disable=SC2030,SC2031
+        export OUTPUT_FORMAT=helm
+        deploy_stackrox >&3 # This is doing an `helm upgrade --install ...` under the hood.
+    )
+
+    # Verify that Scanner v2 and v4 are up.
+    verify_scannerV2_deployed "stackrox"
+    verify_scannerV4_deployed "stackrox"
+}
+
+@test "Fresh installation of HEAD Helm chart with Scanner v4 disabled" {
+    MAIN_IMAGE_TAG=""
+    info "Installing StackRox using HEAD Helm chart with Scanner v4 disabled"
+    if [[ -n "${CURRENT_MAIN_IMAGE_TAG:-}" ]]; then
+        MAIN_IMAGE_TAG=$CURRENT_MAIN_IMAGE_TAG
+        info "Overriding MAIN_IMAGE_TAG=$CURRENT_MAIN_IMAGE_TAG"
+    fi
+    (
+        # shellcheck disable=SC2030,SC2031
+        export MAIN_IMAGE_TAG
+        # shellcheck disable=SC2030,SC2031
+        export ROX_SCANNER_V4_ENABLED=false
+        # shellcheck disable=SC2030,SC2031
+        export OUTPUT_FORMAT=helm
+        deploy_stackrox >&3
+    )
+    verify_scannerV2_deployed "stackrox"
+    verify_no_scannerV4_deployed "stackrox"
```

Comment:
nit
```suggestion
    verify_scannerV4_not_deployed "stackrox"
```

Reply:
hope it's fine to keep the naming at least for now, I don't have strong feelings about it.